src/components/queryBuilder/ModeSwitch.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { RadioButtonGroup, InlineFormLabel } from '@grafana/ui';
+
+export interface ModeSwitchProps {
+  labelA: string;
+  labelB: string;
+  value: boolean;
+  onChange: (value: boolean) => void;
+  label: string;
+  tooltip: string;
+};
+
+/**
+ * Component for switching between modes. Boxes are labled unlike regular Switch.
+ */
+export const ModeSwitch = (props: ModeSwitchProps) => {
+  const { labelA, labelB, value, onChange, label, tooltip } = props;
+
+  const options = [
+    {
+      label: labelA,
+      value: false,
```

Comment:
What if we need more than 2 modes?

Reply:
In the designs there are a few of these switches, but they only ever have 2 modes (the only difference is the labels). If we needed 3 I would change this component to function more like the `QueryTypeSwitcher` (maybe even pass in an array of options). I already have a regular `Switch` component, but its a different style of switch.